src/config/passport.js

```diff
@@ -1,27 +1,30 @@
-const passport = require('passport');
-const LocalStrategy = require('passport-local').Strategy;
-const mongoose = require('mongoose');
+import passport from 'passport';
+import { OAuth2Strategy as GoogleStrategy } from 'passport-google-oauth';
+import { Strategy as FacebookStrategy } from 'passport-facebook';
+import MockStrategy from './mockStrategy';
```

Comment:
Why do we need a mock strategy?

Reply:
It is how we are testing the code written.

That way I can tell passport to use this strategy when we are in a test environment. So my test isn't mocking any function. It is using the actual passport library to test,

That way I can use the same url I use for production, and test the workflow like I am really authenticating through those platforms. It is the reason why our mock strategy is inheriting from `passport.Strategy`.

Reply:
So, the strategy function handles the dual cases of failed authentication (where no user object is returned), and a successful one (where a user object is returned).